acme/acme/messages.py

```diff
@@ -483,3 +484,37 @@ class Revocation(jose.JSONObjectWithFields):
     certificate = jose.Field(
         'certificate', decoder=jose.decode_cert, encoder=jose.encode_cert)
     reason = jose.Field('reason')
+
+
+class Order(ResourceBody):
+    """Order Resource Body.
+
+    :ivar buffer csr: CSR in pem format.
+    :ivar string status:
+    :ivar list of string authorizations:
+    :ivar datetime.datetime expires:
+
+    """
+    identifiers = jose.Field('identifiers', omitempty=True)
+    status = jose.Field('status', omitempty=True, default=None)
+    authorizations = jose.Field('authorizations', omitempty=True)
+    certificate = jose.Field('certificate', omitempty=True)
+    finalize = jose.Field('finalize', omitempty=True)
+    expires = fields.RFC3339Field('expires', omitempty=True)
+
+class OrderResource(ResourceWithURI):
+    """Order Resource.
+
+    :ivar acme.messages.Order body:
+
+    """
+    body = jose.Field('body', decoder=Order.from_json)
+    csr_pem = jose.Field('csr_pem', omitempty=True)
+    authorizations = jose.Field('authorizations')
+    fullchain_pem = jose.Field('fullchain_pem', omitempty=True)
```

Comment:
Do all of these attributes make sense on an `OrderResource` when compared to the other *Resources in this file?

Reply:
Yes? I'm not sure how to evaluate that.

Reply:
Sorry. I just made this as a quick comment without diving into it since it was getting late on Friday night, but I'm not sure if it does match the other resources.

I certainly could be off base as the vast majority of this file hasn't changed since before I was involved in the project, but my understanding of all of the classes here is for each object in the ACME protocol you have a class that defines that object and a `ObjectResource` class that contains the object as the `body` and the other fields are the various `link` headers provided at that endpoint.

The only exception to this which I couldn't figure out is `CertificateResource` where the inclusion of `authzs` doesn't make sense to me, although looking through the git history, it was added when you the ACME spec said to send the CSR and list of authz to the `new-cert` endpoint.

Reply:
We talked about this offline and I'm fine with the changes here.